moveit_planners/chomp/chomp_motion_planner/src/chomp_planner.cpp

```diff
@@ -156,16 +156,78 @@ bool ChompPlanner::solve(const planning_scene::PlanningSceneConstPtr& planning_s
-  if (!optimizer.isInitialized())
+  int replan_count = 0;
+  bool replan_flag = false;
+  double org_learning_rate = 0.04, org_ridge_factor = 0.0, org_planning_time_limit = 10;
+  int org_max_iterations = 200;
+
+  // storing the initial chomp parameters values
+  org_learning_rate = params.learning_rate_;
+  org_ridge_factor = params.ridge_factor_;
+  org_planning_time_limit = params.planning_time_limit_;
+  org_max_iterations = params.max_iterations_;
+
+  ChompOptimizer* optimizer;
+
+  // create a non_const_params variable which stores the non constant version of the const params variable
+  ChompParameters non_constant = non_constant.getNonConstantParams(params);
+
+  // do while loop for replanning (recovery behaviour) if collision free optimized solution not found
+  do
   {
-    ROS_ERROR_STREAM_NAMED("chomp_planner", "Could not initialize optimizer");
-    res.error_code.val = moveit_msgs::MoveItErrorCodes::PLANNING_FAILED;
-    return false;
-  }
-  ROS_DEBUG_NAMED("chomp_planner", "Optimization took %f sec to create", (ros::WallTime::now() - create_time).toSec());
+    if (replan_flag)
+    {
+      non_constant.learning_rate_ += 0.02;     // increase learning rate in hope to find a successful path
+      non_constant.ridge_factor_ += 0.002;     // increase ridge factor to avoid obstacles
+      non_constant.planning_time_limit_ += 5;  // add 5 additional secs in hope to find a solution
+      non_constant.max_iterations_ += 50;      // increase maximum iterations
+    }
+
+    optimizer = new ChompOptimizer(&trajectory, planning_scene, req.group_name, &non_constant, start_state);
+    if (!optimizer->isInitialized())
+    {
+      ROS_ERROR_STREAM_NAMED("chomp_planner", "Could not initialize optimizer");
+      res.error_code.val = moveit_msgs::MoveItErrorCodes::PLANNING_FAILED;
+      return false;
+    }
+
+    ROS_DEBUG_NAMED("chomp_planner", "Optimization took %f sec to create",
+                    (ros::WallTime::now() - create_time).toSec());
+
+    bool optimization_result = optimizer->optimize();
+
+    // replan with updated parameters if no solution is found
+    if (non_constant.enable_failure_recovery_)
+    {
+      // if (!optimization_result && replan_count != non_constant.getMaxRecoveryAttempts())
+      //{
+      ROS_INFO("Planned with Chomp Parameters (learning_rate, ridge_factor, planning_time_limit, "
+               "max_iterations), attempt: # %d ",
+               (replan_count + 1));
+      ROS_INFO("Learning rate: %f ridge factor: %f planning time limit: %f max_iterations %d ",
+               non_constant.learning_rate_, non_constant.ridge_factor_, non_constant.planning_time_limit_,
+               non_constant.max_iterations_);
+      //}
+      if (!optimization_result && replan_count < non_constant.max_recovery_attempts_)
+      {
+        replan_count++;
+        replan_flag = true;
+      }
+      else
+      {
+        break;
+      }
+    }
+    else
+      break;
+  } while (true);
```

Comment:
no need for ``do... while`` if its always ``true``. simply just use ``while`` loop